src/Arcus.Security.Tests.Integration/KeyVault/KeyVaultSecretProviderTests.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Arcus.Security.Core.Exceptions;
+using Arcus.Security.KeyVault;
+using Arcus.Security.KeyVault.Factories;
+using Arcus.Security.Tests.Integration.Logging;
+using Microsoft.Extensions.Configuration;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Arcus.Security.Tests.Integration.KeyVault
+{
+    [Trait(name: "Category", value: "Integration")]
+    public class KeyVaultSecretProviderTests
+    {
+        // The same tests should be tested with different KeyVaultClientFactories 
```

Comment:
I personally would create dedicated methods for those given they are testing different things.
The alternative is passing them via a test parameter.

I think for all tests we should pass them via test parameter, but have dedicated ones for happy path as well given you'd want to test the factory itself as well.

Reply:
agree